config/routes.rb

```diff
@@ -3,5 +3,9 @@
 
   devise_for :users
 
-  resources :events, only: %i[new create update destroy]
+  resources :events, only: %i[new create update destroy] do
+    resources :bookings, only: %i[create update] do
+      patch :cancelation, on: :member
+    end
+  end
```

Comment:
Can you rename your member route "cancel" and give it an alias of "cancel_booking" please ?

Reply:
```
  resources :events, only: %i[create update destroy] do
    resources :bookings, only: %i[create update] do
      patch :cancel_booking, on: :member
    end
  end
```

Reply:
`resources :bookings, only: %i[create] do` doesn't have the `update` action anymore because it was deleted

Reply:
To give an alias to a route you need to use `as: :cancel_booking` before your `on`